src/app/modules/main/communities/controller.nim

```diff
@@ -154,3 +161,9 @@ proc isCommunityRequestPending*(self: Controller, communityId: string): bool =
 
 proc getStatusForContactWithId*(self: Controller, publicKey: string): StatusUpdateDto =
   return self.contactsService.getStatusForContactWithId(publicKey)
+
+proc requestExtractDiscordChannelsAndCategories*(self: Controller, filesToImport: seq[string]) =
+  self.communityService.requestExtractDiscordChannelsAndCategories(filesToImport)
+
+proc extractDiscordChannelsAndCategories*(self: Controller, filesToImport: seq[string]) =
+  self.communityService.extractDiscordChannelsAndCategories(filesToImport)
```

Comment:
This one is a synchronous API, while the one above is an async API. There's a chance the async one gets renamed. See https://github.com/status-im/status-go/pull/2752#discussion_r919879210 for more info

Atm, the synchronous API is no longer used by desktop, so we might as well remove it here, however, as you can see in the linked discussion above, @cammellos advocates for avoiding async APIs, keeping them sync in status-go and then let the client (Status Desktop) wrap it to make it async.

I haven't gone with an async wrapper in this case because the threadpool APIs in nim are quite cumbersome. 

So `requestExtractDiscordCHannelsAndCategories` will cause a signal that can be reacted to asynchronously.